inst/include/interface/place_holder0.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef COMMON_HPP
+#define COMMON_HPP
+
+//#define TMB_MODEL
+
+#ifdef TMB_MODEL
+
+#include <TMB.hpp>
+//#include <Rcpp.h>
+//using namespace Rcpp;
+
+template<typename Type>
+struct model_traits{
+ typedef typename CppAD::vector<Type> data_vector;
+ typedef typename CppAD::vector<Type> parameter_vector;
+};
+
+template<typename T>
+T exp(const T& x){
+  return exp(x);
+}
+
+template <class T>
+const T log(const T& x){return std::log(x);}
+
+#endif
+
+#ifdef STD_LIB
+
+#include <cmath>
+#include <vector>
+
+template<typename Type>
+struct model_traits{
+ typedef typename std::vector<Type> data_vector;
+ typedef typename std::vector<Type> parameter_vector;
+};
+
+template<typename T>
+T exp(const T& x){
+  return std::exp(x);
+}
+
+template <class T>
+const T log(const T& x){return std::log(x);}
```

Comment:
Do we need to define this twice (lines 18 - 24 and lines 39 - 45 are identical to my eye)?

Reply:
Good question, I think the lines 18-24 and lines 39-45 are used for different libraries (TMB v.s. std). @Andrea-Havron-NOAA perhaps has a better explanation than me. I copied her FIMS-software [Rcpp code here](https://github.com/Andrea-Havron-NOAA/FIMS-software/blob/master/src/Rcpp/Common.hpp).